realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -625,6 +625,11 @@ private static void initializeRealm(Realm realm) {
         if (clazz == null || inputStream == null) {
             return null;
         }
+        Table table = getTable(clazz);
+        if (table.hasPrimaryKey()) {
+            throw new UnsupportedOperationException("Cannot create RealmObjects with primary keys using Streams. " +
+                    "Use createOrUpdateObjectFromJson() instead.");
```

Comment:
Or loop with the stream two rounds, find out the primary key in the first round, also document to suggest always put the primary key as the first one if user wants to use JSONStream.... silly solution...